mysql/resource_rds_config_test.go

```diff
@@ -0,0 +1,205 @@
+			}
+		}
+		return nil
+	}
+}
+
+func TestAccResourceRDSConfigChange(t *testing.T) {
+	rName := "test_update"
+	fullResourceName := fmt.Sprintf("mysql_rds_config.%s", rName)
+	binlog := 24
+	binlogUpdated := 48
+	targetDelay := 3200
+	targetDelayUpdated := 7400
+
+	ctx := context.Background()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccRDSCheckDestroy(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccRDSConfig_basic(rName, binlog, targetDelay),
+				Check: resource.ComposeTestCheckFunc(
+					testAccRDSConfigExists(fmt.Sprintf("mysql_rds_config.%s", rName)),
+				),
+			},
+			{
+				ResourceName: fullResourceName,
+				ImportState:  true,
+				PreConfig: func() {
+					db, err := connectToMySQL(ctx, testAccProvider.Meta().(*MySQLConfiguration))
+					if err != nil {
+						return
+					}
+
+					_, err = db.QueryContext(ctx, fmt.Sprintf("call mysql.rds_set_configuration('binlog retention hours', %d)", binlogUpdated))
+					if err != nil {
+						fmt.Errorf("%v", err)
+					}
+
+					_, err = db.QueryContext(ctx, fmt.Sprintf("call mysql.rds_set_configuration('target delay', %d)", targetDelayUpdated))
+					if err != nil {
+						fmt.Errorf("%v", err)
+					}
+				},
+				Check: resource.ComposeTestCheckFunc(
+					testAccRDSConfigExists(fmt.Sprintf("mysql_rds_config.%s", rName)),
+					testAccRDSCheck_full(fullResourceName, binlog, targetDelay),
+				),
+			},
+		},
+	})
+}
+
+func testAccRDSCheck_full(rn string, binlog int, targetDelay int) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[rn]
+		if !ok {
+			return fmt.Errorf("resource not found: %s", rn)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("RDS config id not set")
+		}
+
+		ctx := context.Background()
+		db, err := connectToMySQL(ctx, testAccProvider.Meta().(*MySQLConfiguration))
+		if err != nil {
+			return err
+		}
+
+		stmtSQL := "call mysql.rds_show_configuration"
+
+		rows, err := db.QueryContext(ctx, stmtSQL)
+		if err != nil {
+			return err
+		}
+
+		results := make(map[string]string)
+		for rows.Next() {
+			var name, description string
+			var value sql.NullString
+
+			if err := rows.Scan(&name, &value, &description); err != nil {
+				return fmt.Errorf("failed reading RDS config: %v", err)
+			}
+
+			if value.Valid {
+				results[name] = value.String
+			} else {
+				results[name] = "0"
+			}
+		}
+
+		binlog_retention_period, err := strconv.Atoi(results["binlog retention hours"])
+		if err != nil {
+			return fmt.Errorf("failed reading RDS config: %v", err)
+		}
+		replication_target_delay, err := strconv.Atoi(results["target delay"])
+		if err != nil {
+			return fmt.Errorf("failed reading RDS config: %v", err)
+		}
+
+		if binlog_retention_period == binlog {
+			return fmt.Errorf("binlog retention should NOT be %d", binlog)
```

Comment:
So what should it be and what it is? Could you add it to the message?

Reply:
Will fix it